packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,150 @@
+import { Template, Match } from '../../../assertions';
+import * as logs from '../../../aws-logs';
+import * as s3 from '../../../aws-s3';
+import * as s3deploy from '../../../aws-s3-deployment';
+import * as cdk from '../../../core';
+import { CustomResourceConfig } from '../../lib/custom-resource-config/custom-resource-config';
+
+describe('when logging is undefined', () => {
+  test('when CustomResourceConfig.addLogRetentionLifetime is called, singleton-backed custom resource has new logGroup with retention period specified', () => {
+    const customResourceLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(customResourceLogRetention);
+    const stack = new cdk.Stack(app);
+
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      LoggingConfig: {
+        LogGroup: {
+          Ref: Match.stringLikeRegexp('CustomCDKBucketDeployment'),
```

Comment:
can we avoid using `stringLikeRegexp`? The correct result here is not a string containing that thing, it exactly the string 'CustomCDKBucketDeployment'.  You should be able to say:

```
Ref: 'CustomCDKBucketDeployment',
```